tests/dvslib/dvs_mirror.py

```diff
@@ -0,0 +1,100 @@
+        self.config_db.create_entry("MIRROR_SESSION", name, mirror_entry)
+
+    def create_erspan_session(self, name, src, dst, gre, dscp, ttl, queue, policer=None, src_ports=None, direction="BOTH"):
+        mirror_entry = {"type": "ERSPAN"}
+        mirror_entry["src_ip"] = src
+        mirror_entry["dst_ip"] = dst
+        mirror_entry["gre_type"] = gre
+        mirror_entry["dscp"] = dscp
+        mirror_entry["ttl"] = ttl
+        mirror_entry["queue"] = queue
+        if policer:
+            mirror_entry["policer"] = policer
+        if src_ports:
+            mirror_entry["src_port"] = src_ports
+        mirror_entry["direction"] = direction
+
+        self.config_db.create_entry("MIRROR_SESSION", name, mirror_entry)
+
+    def remove_mirror_session(self, name):
+        self.config_db.delete_entry("MIRROR_SESSION", name)
+
+    def verify_no_mirror(self):
+        self.config_db.wait_for_n_keys("MIRROR_SESSION", 0)
+        self.state_db.wait_for_n_keys("MIRROR_SESSION_TABLE", 0)
+
+    def verify_session_status(self, name, status="active", expected=1):
+        self.state_db.wait_for_n_keys("MIRROR_SESSION_TABLE", expected)
+        if expected:
+            self.state_db.wait_for_field_match("MIRROR_SESSION_TABLE", name, {"status": status})
+
+    def verify_port_mirror_config(self, dvs, ports, direction, session_oid="null"):
+        fvs = dvs.counters_db.get_entry("COUNTERS_PORT_NAME_MAP", "")
+        fvs = dict(fvs)
+        for p in ports:
+            port_oid = fvs.get(p)
+            member = dvs.asic_db.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_PORT", port_oid)
+            if direction in {"RX", "BOTH"}:
+                assert member["SAI_PORT_ATTR_INGRESS_MIRROR_SESSION"] == "1:"+session_oid
+            else:
+                assert "SAI_PORT_ATTR_INGRESS_MIRROR_SESSION" not in member.keys() or member["SAI_PORT_ATTR_INGRESS_MIRROR_SESSION"] == "0:null"
+            if direction in {"TX", "BOTH"}:
+                assert member["SAI_PORT_ATTR_EGRESS_MIRROR_SESSION"] == "1:"+session_oid
+            else:
+                assert "SAI_PORT_ATTR_EGRESS_MIRROR_SESSION" not in member.keys() or member["SAI_PORT_ATTR_EGRESS_MIRROR_SESSION"] == "0:null"
+
+    def verify_session_db(self, dvs, name, asic_table=None, asic=None, state=None, asic_size=None):
+        if asic:
+            fv_pairs = dvs.asic_db.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_MIRROR_SESSION", asic_table)
+            assert all(fv_pairs.get(k) == v for k, v in asic.items())
+            if asic_size:
+                assert asic_size == len(fv_pairs)
+        if state:
+            fv_pairs = dvs.state_db.get_entry("MIRROR_SESSION_TABLE", name)
+            assert all(fv_pairs.get(k) == v for k, v in state.items())
+
+    def verify_session_policer(self, dvs, policer_oid, cir):
+        if cir:
+            entry = dvs.asic_db.get_entry("ASIC_STATE:SAI_OBJECT_TYPE_POLICER", policer_oid)
+            assert entry["SAI_POLICER_ATTR_CIR"] == cir
```

Comment:
I would probably still use `wait` methods for these even if the polling is covered by other methods in the existing code. If someone tries to use one of these methods elsewhere it may be difficult to diagnose the failures. :)

Reply:
Ok. makes sense. Will do. I thought if the session is active, rest all will be in place, but as u said, if someone else is using then it may cause failures.

Reply:
Done. Have taken care.